helixlauncher-core/src/launch/prepared.rs

```diff
@@ -317,21 +313,13 @@ pub async fn prepare_launch(
     })
 }
 
-pub async fn version_exists(path: String, version: String) -> bool {
-    let response = reqwest::get(format!("{META}{path}/index.json"))
-        .await
-        .expect("an error occurred while fetching data from meta");
-
-    let index: Index = serde_json::from_str(
-        response
-            .text()
-            .await
-            .expect("error while reading body")
-            .as_str(),
-    )
-    .expect("error while converting to json");
+pub async fn version_exists(path: String, version: String, config: &Config) -> bool {
     let mut found: bool = false;
-    for item in index {
+    for item in HelixLauncherMeta::new(config)
+        .get_component_index(&path)
+        .await
+        .expect("Error while checking if version exists")
```

Comment:
I don't think this is currently used anywhere, but panicking on non-logic errors is bad... not really sure if it _will_ ever be used though (if it is, it'd go in the meta module)

Reply:
This was the old behavior so I kept it. I can do some more refactoring if you want, too